### PR TITLE
migrate base TestCase class and other supporting functions from apps/amo

### PR DESCRIPTION
**Pull request by eviljeff**
Largely a find&replace, though I removed a couple of unused functions and reworked the imports for the tests that initially failed.

### CODE REVIEW

File: mkt/files/tests/test_helpers.py [hunk 1]
```diff
@@ -272,8 +272,9 @@ def change(self, file, text, filename='install.js'):
         open(path, 'w').write(data)
 
 
-class TestSafeUnzipFile(amo.tests.TestCase, amo.tests.AMOPaths):
+class TestSafeUnzipFile(mkt.site.tests.TestCase, mkt.site.tests.MktPaths):
 
+    ##(ej)## is any of this actually used? Its all testing for addons files
```
Comment (eviljeff) [inline mkt/files/tests/test_helpers.py]: Do we need to be testing these anymore?  Some seem redundant at least.

File: mkt/site/tests/__init__.py [hunk 2]
```diff
@@ -59,3 +122,870 @@ def test_to_dict(self):
...
+
+
+class MktPaths(object):
+    """Mixin for getting common Marketplace Paths."""
+
+    def file_fixture_path(self, name):
+        path = 'mkt/files/fixtures/files/%s' % name
+        return os.path.join(settings.ROOT, path)
+
+    ##(ej)## only used in mkt/files/tests/test_helpers and for addons files.
```
Comment (eviljeff) [inline mkt/site/tests/__init__.py]: linked to earlier question.  Seems entirely redundant.

File: mkt/submit/tests/test_views.py [hunk 3]
```diff
@@ -1011,7 +1010,8 @@ def test_done(self):
         eq_(res.status_code, 200)
 
 
-class TestNextSteps(amo.tests.TestCase):
+class TestNextSteps(TestCase):
+    ##(ej)## Delete below now?
```
Comment (eviljeff) [inline mkt/submit/tests/test_views.py]: IARC is deployed.  Delete?

File: mkt/files/tests/test_helpers.py (hunk 1 above)
Comment (diox) [inline mkt/files/tests/test_helpers.py]: Yes, we should. It should be updated to use a packaged app instead, but it's definitely useful.

File: mkt/site/tests/__init__.py (hunk 2 above)
Comment (diox) [inline mkt/site/tests/__init__.py]: See answer above.

File: mkt/submit/tests/test_views.py (hunk 3 above)
Comment (diox) [inline mkt/submit/tests/test_views.py]: I think the test is still useful, despite what the comment says (and the comment should be removed).

What do you think, @ngokevin ?

File: lib/crypto/tests.py [hunk 4]
```diff
@@ -12,7 +12,7 @@
 from nose.tools import eq_, raises
 from requests import Timeout
 
-import amo.tests
+import mkt.site.tests
```
Comment (robhudson) [inline lib/crypto/tests.py]: l < m

File: mkt/reviewers/tests/test_views.py [hunk 5]
```diff
@@ -49,6 +47,8 @@
                                  route_reviewer)
 from mkt.site.fixtures import fixture
 from mkt.site.helpers import absolutify, isotime
+from mkt.site.tests import (app_factory, check_links, days_ago, formset, initial,
+                       req_factory_factory, user_factory, version_factory)
```
Comment (robhudson) [inline mkt/reviewers/tests/test_views.py]: Please fix line wrapping

File: mkt/api/tests/test_forms.py [hunk 6]
```diff
@@ -6,12 +6,12 @@
 from nose.tools import eq_, ok_
 
 import amo
```
Comment (robhudson) [inline mkt/api/tests/test_forms.py]: `import amo` no longer needed?

File: mkt/comm/tests/test_serializers.py [hunk 7]
```diff
@@ -3,32 +3,32 @@
 from nose.tools import eq_, ok_
 
 import amo
-import amo.tests
```
Comment (robhudson) [inline mkt/comm/tests/test_serializers.py]: `import amo` no longer needed?

**Comment by robhudson** [comment]
Running flake8 on this gives the following, some of which aren't your fault but if you're touching the files you might be inclined to fix them? :)

```
mkt/api/tests/__init__.py:33:1: W391 blank line at end of file
mkt/api/tests/test_forms.py:8:1: F401 'amo' imported but unused
mkt/comm/tests/test_forms.py:11:4: E111 indentation is not a multiple of four
mkt/comm/tests/test_forms.py:12:8: E111 indentation is not a multiple of four
mkt/comm/tests/test_forms.py:14:4: E111 indentation is not a multiple of four
mkt/comm/tests/test_forms.py:15:8: E111 indentation is not a multiple of four
mkt/comm/tests/test_forms.py:21:8: E111 indentation is not a multiple of four
mkt/comm/tests/test_forms.py:22:8: E111 indentation is not a multiple of four
mkt/comm/tests/test_forms.py:24:4: E111 indentation is not a multiple of four
mkt/comm/tests/test_forms.py:25:8: E111 indentation is not a multiple of four
mkt/comm/tests/test_forms.py:26:8: E111 indentation is not a multiple of four
mkt/comm/tests/test_forms.py:27:8: E111 indentation is not a multiple of four
mkt/comm/tests/test_forms.py:28:8: E111 indentation is not a multiple of four
mkt/comm/tests/test_forms.py:29:8: E111 indentation is not a multiple of four
mkt/comm/tests/test_forms.py:31:4: E111 indentation is not a multiple of four
mkt/comm/tests/test_forms.py:32:8: E111 indentation is not a multiple of four
mkt/comm/tests/test_forms.py:33:8: E111 indentation is not a multiple of four
mkt/comm/tests/test_forms.py:34:8: E111 indentation is not a multiple of four
mkt/comm/tests/test_serializers.py:1:1: F401 'settings' imported but unused
mkt/comm/tests/test_serializers.py:5:1: F401 'amo' imported but unused
mkt/comm/tests/test_serializers.py:10:1: F401 'comm' imported but unused
mkt/api/tests/test_fields.py:11:1: F401 'amo' imported but unused
mkt/comm/tests/test_utils_mail.py:15:1: F401 'get_recipients' imported but unused
mkt/constants/tests/test_regions.py:2:1: F401 'translation' imported but unused
mkt/commonplace/tests/test_views.py:10:1: F401 'escape' imported but unused
mkt/commonplace/tests/test_views.py:45:1: E302 expected 2 blank lines, found 1
mkt/developers/tests/test_helpers.py:125:55: E251 unexpected spaces around keyword / parameter equals
mkt/developers/tests/test_helpers.py:136:55: E251 unexpected spaces around keyword / parameter equals
mkt/comm/tests/test_views.py:5:1: F401 'mail' imported but unused
mkt/developers/tests/test_views_ownership.py:134:9: F841 local variable 'r' is assigned to but never used
mkt/developers/tests/test_views_ownership.py:166:9: F841 local variable 'r' is assigned to but never used
mkt/developers/tests/test_views_validation.py:16:1: F401 'amo' imported but unused
mkt/feed/tests/test_fields.py:9:1: F401 'serializers' imported but unused
mkt/feed/tests/test_fields.py:13:1: F401 'feed' imported but unused
mkt/feed/tests/test_fields.py:15:1: F401 'FeedTestMixin' imported but unused
mkt/feed/tests/test_fields.py:16:1: F401 'WebappIndexer' imported but unused
mkt/feed/tests/test_serializers.py:7:1: F401 'amo' imported but unused
mkt/files/tests/test_helpers.py:277:5: E265 block comment should start with '# '
mkt/files/tests/test_utils_.py:107:5: E303 too many blank lines (2)
mkt/operators/tests/test_views.py:8:1: F401 'amo' imported but unused
mkt/prices/tests/test_utils.py:20:1: E302 expected 2 blank lines, found 1
mkt/reviewers/tests/test_forms.py:7:1: F401 'amo' imported but unused
mkt/search/tests/test_indexer.py:6:1: E302 expected 2 blank lines, found 1
mkt/reviewers/tests/test_tasks.py:11:1: F401 'AddonPurchase' imported but unused
mkt/reviewers/tests/test_tasks.py:11:1: F401 'Refund' imported but unused
mkt/reviewers/tests/test_tasks.py:12:1: F401 'Contribution' imported but unused
mkt/reviewers/tests/test_tasks.py:16:1: F401 'UserProfile' imported but unused
mkt/ratings/tests/test_views.py:218:16: E713 test for membership should be 'not in'
mkt/site/tests/test_fields.py:9:1: F401 'amo' imported but unused
mkt/site/tests/test_helpers.py:11:1: F401 'amo' imported but unused
mkt/site/tests/test_mail.py:1:1: F401 're' imported but unused
mkt/feed/tests/test_views.py:1276:9: F841 local variable 'data' is assigned to but never used
mkt/feed/tests/test_views.py:1281:9: F841 local variable 'data' is assigned to but never used
mkt/feed/tests/test_views.py:1290:9: F841 local variable 'data' is assigned to but never used
mkt/site/tests/test_middleware.py:366:5: E124 closing bracket does not match visual indentation
mkt/site/tests/__init__.py:9:1: F401 'wraps' imported but unused
mkt/site/tests/__init__.py:19:1: F401 'SimpleCookie' imported but unused
mkt/site/tests/__init__.py:669:5: E265 block comment should start with '# '
mkt/submit/tests/test_models.py:3:1: F401 'amo' imported but unused
mkt/submit/tests/test_models.py:10:1: E302 expected 2 blank lines, found 1
mkt/translations/tests/test_forms.py:13:5: E301 expected 1 blank line, found 0
mkt/site/tests/test_views.py:13:1: F401 'amo' imported but unused
mkt/translations/tests/test_helpers.py:8:1: F401 'amo' imported but unused
mkt/translations/tests/test_utils.py:27:29: W291 trailing whitespace
mkt/users/tests/test_models.py:9:1: F401 'amo' imported but unused
mkt/users/tests/test_views.py:4:1: F401 'urllib' imported but unused
mkt/users/tests/test_views.py:7:1: F401 'override_settings' imported but unused
mkt/users/tests/test_views.py:9:1: F401 'mock' imported but unused
mkt/users/tests/test_views.py:14:1: F401 'amo' imported but unused
mkt/submit/tests/test_views.py:1014:5: E265 block comment should start with '# '
mkt/webapps/tasks.py:136:5: E129 visually indented line with same indent as next logical line
```

File: mkt/files/tests/test_helpers.py (hunk 1 above)
Comment (eviljeff) [inline mkt/files/tests/test_helpers.py]: /me will rewrite to use packaged apps

File: mkt/submit/tests/test_views.py (hunk 3 above)
Comment (eviljeff) [inline mkt/submit/tests/test_views.py]: I removed my comment and left @ngokevin 's for now.  If a decision is made prior to this being ready to merge I'll change but doesn't seem like a blocker.

**Comment by eviljeff** [comment]
(made changes + broke&fixed a few tests)
r?

File: mkt/files/tests/test_helpers.py [hunk 8]
```diff
@@ -259,53 +260,58 @@ def test_diffable_one_directory(self):
...
 
     def test_is_broken(self):
-        zip = SafeUnzip(self.xpi_path('signed'))
+        zip = SafeUnzip(self.packaged_app_path('signed.zip'))
         zip.is_valid()
-        zip.info[2].filename = 'META-INF/foo.sf'
+        sf_re = re.compile('^META\-INF/(\w+)\.sf$')
+        for info in zip.info:
+            if sf_re.match(info.filename):
+                info.filename= 'META-INF/foo.foo'
```
Comment (robhudson) [inline mkt/files/tests/test_helpers.py]: Spaces around the `=`

**Comment by robhudson** [comment]
r+wc

**Pull request merged**